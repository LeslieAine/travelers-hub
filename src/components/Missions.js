const Missions = () => (
  <div>My Profile</div>
);

export default Missions;
