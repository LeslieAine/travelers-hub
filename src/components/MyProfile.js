const MyProfile = () => (
  <div>My Profile</div>
);

export default MyProfile;
