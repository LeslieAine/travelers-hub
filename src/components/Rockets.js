const Rockets = () => (
  <div>My Profile</div>
);

export default Rockets;
