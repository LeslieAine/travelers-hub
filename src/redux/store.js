import { configureStore } from '@reduxjs/toolkit';
import rocketsReducer from './rockets/rocketsSlice';
import missionsReducers from './missions/missionsSlice';

const store = configureStore({
  reducer: {
    rockets: rocketsReducer,
    missions: missionsReducers,
  },
});

export default store;
