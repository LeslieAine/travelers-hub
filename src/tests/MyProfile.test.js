import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import '@testing-library/jest-dom/extend-expect';
import TestRenderer, { act } from 'react-test-renderer';
import Rockets from '../components/Rockets';
import Missions from '../components/Missions';
import MyProfile from '../components/MyProfile';

import store from '../redux/store';

describe('Profile', () => {
  test('Render MyProfile Component', () => {
    const tree = TestRenderer.create(
      <Provider store={store}>
        <Missions />
        <Rockets />
        <MyProfile />
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Rocket Button', async () => {
    render(
      <Provider store={store}>
        <Rockets />
        {/* <Missions /> */}
      </Provider>,
    );

    const reserveBtn = await screen.findAllByText('Reserve Rocket');
    expect(reserveBtn).toHaveLength(4);
  });

  test('Reserved Rocket name should be in profile', async () => {
    render(
      <Provider store={store}>
        <Rockets />
        {/* <Missions /> */}
        <MyProfile />
      </Provider>,
    );

    act(() => {
      const reserveBtn = screen.getAllByText('Reserve Rocket');
      reserveBtn[0].click();
    });

    const rocketName = await screen.getAllByText('Falcon 1');
    expect(rocketName[1]).toBeInTheDocument();
  });
});
