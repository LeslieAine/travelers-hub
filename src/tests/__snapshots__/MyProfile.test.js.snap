// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Render MyProfile Component 1`] = `
Array [
  <p>
    pending...
  </p>,
  <div
    className="container"
  >
    <div />
    <div
      className="rocket"
    >
      <h2>
        My Rockets
      </h2>
      <table
        className="table"
      >
        <tbody
          className="tbody"
        />
      </table>
    </div>
  </div>,
]
`;
